### PR TITLE
Add groupList to custom extension in standard X509 cert.

### PR DESCRIPTION
**Pull request by rgooch**

### CODE REVIEW

File: lib/certgen/certgen.go [hunk 1]
```diff
@@ -260,13 +263,47 @@ func genSANExtension(userName string, kerberosRealm *string) (*pkix.Extension, e
...
+	encodedValue, err := asn1.Marshal(groups)
+	if err != nil {
+		return nil, err
+	}
+	oid, err := stringToIntSlice(constants.GroupListOID)
+	if err != nil {
+		return nil, err
+	}
+	groupListExtension := pkix.Extension{
+		Id:    oid,
```
Comment (cviecco) [inline lib/certgen/certgen.go]: I dont like the import of all dominator constants here. Just add a local constant of the oid string text. You remove 1 dependency and about 20 lines of code.

File: lib/certgen/certgen_test.go [hunk 2]
```diff
@@ -281,20 +282,35 @@ func derBytesCertToCertAndPem(derBytes []byte) (*x509.Certificate, string, error
 	return cert, pemCert, nil
 }
 
-//Genx509SCert(userName string, userPubkey string, caCertString string, caPrivateKeyString string)
+//GenxUser509Cert(userName string, userPubkey string, caCertString string, caPrivateKeyString string)
```
Comment (cviecco) [inline lib/certgen/certgen_test.go]: minor: is GnUserx509Cert

File: lib/certgen/certgen.go (hunk 1 above)
Comment (rgooch) [reply lib/certgen/certgen.go]: Grumble. It's like writing the same value into different databases.
Anyway, dependency removed.

File: lib/certgen/certgen_test.go (hunk 2 above)
Comment (rgooch) [reply lib/certgen/certgen_test.go]: Fixed.

**Comment by rgooch** [comment]
Group list information no longer added to X509 certificate by default. PTAL.

**Comment by rgooch** [comment]
Ping? Comments have been addressed.

**Comment by cviecco** [comment]
Is better now, but somehow I dont like the parsing of the add-Groups in that function and not in the caller. (And not the default behavior has not changed). Any suggestions or should we leave it as is?

**Comment by rgooch** [comment]
I'm not sure I understand your point. Do you mean how `GenUserX509Cert` has a list of groups as a parameter and then builds the extensions? Given that this function builds the cert, I think it makes sense for it to build the extensions as well. Also, I don't think the caller wants to have to deal with that. You would want a wrapper function that takes the same parameters as `GenUserX509Cert` has.

**Comment by cviecco** [comment]
It is fine for now, the default behavior has not changes which is my primary concerrn

**Pull request merged**